assets/src/design-system/components/checkbox/index.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { forwardRef } from 'react';
+import propTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+/**
+ * Internal dependencies
+ */
+import { Checkmark } from '../../icons';
+
+const CONTAINER_WIDTH = 24;
+const BORDER_WIDTH = 2;
+
+const CheckboxContainer = styled.div(
+  ({ disabled, theme }) => css`
+    position: relative;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: ${CONTAINER_WIDTH}px;
+    width: ${CONTAINER_WIDTH}px;
+    margin: 8px;
+    border-radius: ${theme.borders.radius.small};
+    border: ${BORDER_WIDTH}px solid ${theme.colors.border.defaultNormal};
+
+    :focus-within {
+      border: ${BORDER_WIDTH}px solid ${theme.colors.border.focus};
+    }
+
+    :active {
+      border: ${BORDER_WIDTH}px solid ${theme.colors.border.defaultNormal};
+      box-shadow: 0 0 0 8px ${theme.colors.shadows.active};
+    }
+
+    ${disabled &&
+    css`
+      border: ${BORDER_WIDTH}px solid ${theme.colors.border.disable};
```

Comment:
nit: since the border width is consistent, you can just update the border-color here and line 48